alpha/apps/kaltura/lib/kEntitlementUtils.php

```diff
@@ -521,4 +521,22 @@ protected static function initCategoryModeration (ks $ks = null)
 		if ($enableCategoryModeration)
 			self::$categoryModeration = true;
 	}
+
+	/**
+	 * @param entry $dbEntry
+	 * @return bool if current user is admin / entry's owner / co-editor
+	 */
+	public static function isEntitledForEditEntry( entry $dbEntry )
+	{
+		if ( kCurrentContext::$is_admin_session || kCurrentContext::getCurrentKsKuserId() == $dbEntry->getKuserId())
+			return true;
+
+		$entitledKusers = baseObjectUtils::getObjectIdsAsArray($dbEntry->getEntitledKusersEdit());
```

Comment:
Simple explode should be good enough, but entry::getEntitledKusersEdit do implode before returning, so it doesn't make send to implode and explode, so I would create a new method called entry::getEntitledKusersEditArray.